### PR TITLE
Navbar and LeftSideBar Refactoring

### LINKED ISSUE #39: Login Styling Change for Smaller Devices
Change the Navbar Styling to look better for smaller devices.

### LINKED ISSUE #44: Refactor Navbar and SideBar to use react-bootstrap and work for smaller devices.
For example, to utilize DropdownMenu

### PR DESCRIPTION
**Pull request by MauroGuerrero20**
# Features
- Utilizes **Material UI** to refactor the styling of the **Navbar** and **LeftSideBar**
- Adds a **RightSideBar** using **material ui**. **Currently commented out**.
- **Fixes some incompatibilities** with @antoniolinhart's pop outs and @steven-solar's clubs
- Nabar and Sidebar now **work with mobile**. Fixes styling issues from before.

# Demo
New Navbar and Sidebar Demo

- On the demo while in mobile debug, the Create List Modal doesn't close the drawer. I have added the ability so it close it automatically.

### CODE REVIEW

File: capstone-frontend/src/components/LeftSideBar.jsx [hunk 1]
```diff
@@ -1,122 +1,454 @@
...
-
-        </ul>
-      </div >
-    );
-  }
-}
+  return (
+    <div id="left-sidebar-navbar" className={classes.root}>
+      <AppBar
+        style={{ backgroundColor: '#007bff' }}
```
Comment (MauroGuerrero20) [inline capstone-frontend/src/components/LeftSideBar.jsx:226]: For some reason, when I try to put this style in ```LeftSideBar.css``` it doesn't actually apply it to the page. Consequently, I decide to just leave it here.

File: capstone-frontend/src/App.css [hunk 2]
```diff
@@ -1,48 +1,12 @@
...
 
-.sidebar-container {
-  margin-top: 56px;
+#right-sidebar {
+  margin-top: 40px;
+  margin-right: -8px;
   min-height: 100vh;   
-  background-color: #333;
+  background-color: whitesmoke;
+  border: 1px solid #ccc
```
Comment (antoniolinhart) [inline capstone-frontend/src/App.css]: nit: alphabetize styles

File: capstone-frontend/src/App.test.jsx [hunk 3]
```diff
@@ -17,21 +15,15 @@ const getUserBookLists = () => {
...
-  render(<Router><Navbar /></Router>, container);
-  const navbar = document.getElementById("navbar");
-  expect(navbar).toBeInTheDocument();
-});
\ No newline at end of file
+// test('renders right sidebar', () => {
+//   render(<Router><RightSideBar />}/></Router>, container);
+//   const rightSidebar = document.getElementById("right-sidebar-container");
+//   expect(rightSidebar).toBeInTheDocument();
+// });
```
Comment (antoniolinhart) [inline capstone-frontend/src/App.test.jsx:33]: Maybe leave a comment here explaining why it is commented out? I believe we decided that we would not have a right sidebar until we implemented the friends feature so there is content to place there?

File: capstone-frontend/src/components/BookDescriptionOverlay.jsx [hunk 4]
```diff
@@ -32,9 +32,13 @@ const popoutDescription = (book) => {
...
-    <OverlayTrigger trigger='click' placement='auto' overlay={popoutDescription(props.book)}>
-      {props.children}
-    </OverlayTrigger>
+    <>
+      {props.book &&
+        <OverlayTrigger trigger='click' placement='auto' overlay={popoutDescription(props.book)}>
+          {props.children}
+        </OverlayTrigger>
+      }
+    </>
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/BookDescriptionOverlay.jsx:41]: Thanks for changing this section! It shouldn't ever be rendered without a correct book prop, but it is good to have the check just in case

File: capstone-frontend/src/App.css [hunk 5]
```diff
@@ -1,48 +1,12 @@
...
-  from {
-    transform: rotate(0deg);
-  }
-  to {
-    transform: rotate(360deg);
-  }
-}
-
-#body-row {
+#main-body {
```
Comment (steven-solar) [inline capstone-frontend/src/App.css:2]: A little worried about the use of magic numbers here, but I know you've tested it on several different screen sizes.

Comment (steven-solar) [reply capstone-frontend/src/App.css:2]: Also it is what we were using before too.

File: capstone-frontend/src/App.jsx [hunk 6]
```diff
@@ -37,8 +24,6 @@ class App extends Component {
 
   fetchBookLists = async () => {
 
-    // TODO(#74): Add Spinner to LeftSide when fetching BookList from App.jsx
```
Comment (steven-solar) [inline capstone-frontend/src/App.jsx:40]: Why did you remove this TODO?

File: capstone-frontend/src/App.test.jsx [hunk 7]
```diff
@@ -17,21 +15,15 @@ const getUserBookLists = () => {
...
-  expect(rightSidebar).toBeInTheDocument();
-});
-
-test('renders nav', () => {
-  render(<Router><Navbar /></Router>, container);
-  const navbar = document.getElementById("navbar");
-  expect(navbar).toBeInTheDocument();
-});
\ No newline at end of file
+// test('renders right sidebar', () => {
```
Comment (steven-solar) [inline capstone-frontend/src/App.test.jsx:29]: Good idea commenting this out rather than deleting it in case we implement later.

File: capstone-frontend/src/BookSearchTile.test.js [hunk 8]
```diff
@@ -12,7 +12,8 @@ document.body.appendChild(container);
 const book = {
   title: 'Title',
   authors: ['Author'],
-  thumbnailLink: 'https://http.cat/100.jpg'
+  thumbnailLink: 'https://http.cat/100.jpg',
```
Comment (steven-solar) [inline capstone-frontend/src/BookSearchTile.test.js:15]: Nice picture! 

image

File: capstone-frontend/src/components/BookDescriptionOverlay.jsx [hunk 9]
```diff
@@ -6,7 +6,7 @@ import { Popover, OverlayTrigger } from 'react-bootstrap';
 // When referenced as a component in the overlay prop, the Popover always appears at the
 // top left of the screen even when there is a predefined placement.
 const popoutDescription = (book) => {
-  let description = book.description || "";
+  let description = (typeof book !== 'undefined') ? book.description : "";
```
Comment (steven-solar) [inline capstone-frontend/src/components/BookDescriptionOverlay.jsx:9]: In what cases is the book type `undefined`?

File: capstone-frontend/src/components/ClubPage.jsx [hunk 10]
```diff
@@ -17,59 +17,61 @@ class ClubPage extends Component {
...
+    this.setState({ members: [] });
     for (let i = 0; i < this.state.club.memberIDs.length; i++) {
       await fetch(`/api/user?id=${this.state.club.memberIDs[i]}`)
-          .then(response => response.json())
-          .then(memberJson => memberJson && this.setState({members: [...this.state.members, memberJson]}))
-          .catch(function(err) {
-            //TODO #61: Centralize error output
-            alert(err); 
-          });
+        .then(response => response.json())
```
Comment (steven-solar) [inline capstone-frontend/src/components/ClubPage.jsx:69]: I know all this formatting is done by Gitpod, but it may create merge conflicts later. Changing unrelated files makes PRs more difficult to review, but I know this styling is automated.

File: capstone-frontend/src/components/CreateClub.jsx [hunk 11]
```diff
@@ -17,10 +17,11 @@ class CreateClub extends Component {
...
     data.ownerID = window.localStorage.getItem("userID");
 
+    console.log("Create club", data)
+
     fetch("/api/clubs", {method: "post", body: JSON.stringify(data)})
-        .then(function() {
-          history.push(`/clubpage/${data.id}`);
-        })
+        .then(resp => resp.json())
+        .then(club => history.push(`/clubpage/${data.id}`))
```
Comment (steven-solar) [inline capstone-frontend/src/components/CreateClub.jsx]: Should this read `/clubpage${club.id}` instead of `/clubpage${data.id}`? I believe this was breaking before because data does not have an id, but the response does (since we let Firebase generate our ids, and place them in the response).

File: capstone-frontend/src/components/LeftSideBar.jsx [hunk 12]
```diff
@@ -1,122 +1,454 @@
...
+              <ListItemText primary="Search" />
+
+            </ListItem>
+          </Link>
+
+          <Link to="/myreads" className="remove-link-style" id="mybooks-link">
+            <ListItem button >
+
+              <ListItemIcon>
+                <MenuBookIcon />
```
Comment (steven-solar) [inline capstone-frontend/src/components/LeftSideBar.jsx:359]: Is it possible to render the names of the tabs rather than icons? I don't think the icons are super clear/obvious to users which tab opens which page.

File: capstone-frontend/src/components/Logout.jsx [hunk 13]
```diff
@@ -22,24 +20,12 @@ export class Logout extends Component {
...
-
-          <Dropdown.Item eventKey="1">
-            <GoogleLogout
-              clientId="962122785123-t0pm10o610q77epuh9d1jjs29hamm1nf.apps.googleusercontent.com"
-              buttonText="Logout"
-              onLogoutSuccess={this.logoutResponseSuccess}
-              isSignedIn={false} />
-          </Dropdown.Item>
-        </DropdownButton>
+      <div style={this.props.loginStyle || {}}>
```
Comment (steven-solar) [inline capstone-frontend/src/components/Logout.jsx:23]: In what case would `<Logout>` not be passed a loginStyle?

File: capstone-frontend/src/components/MyClubs.jsx [hunk 14]
```diff
@@ -16,41 +16,47 @@ class MyClubs extends Component {
...
-                                  //TODO #61: Centralize error output
-                                  alert(err); 
-                                });
+        .then(response => response.json())
+        .catch(function (err) {
+          //TODO #61: Centralize error output
+          alert(err);
+        });
+      let book;
+      if (c.gbookID === '') {
```
Comment (steven-solar) [inline capstone-frontend/src/components/MyClubs.jsx:43]: Nice fix! Glad we figured this out together.

File: capstone-frontend/src/components/SearchBookModal.jsx [hunk 15]
```diff
@@ -60,6 +62,8 @@ export class SearchBookModal extends Component {
   addBookToList = (book) => {
     // Rerender
     this.setState({ addedBooksIDs: [...this.state.addedBooksIDs, book.id], addedBooks: [...this.state.addedBooks, book] })
+
+    console.log(this.state)
```
Comment (steven-solar) [inline capstone-frontend/src/components/SearchBookModal.jsx]: `nit`: remove console logs

File: capstone-frontend/src/components/SearchBookModal.jsx [hunk 16]
```diff
@@ -76,6 +80,8 @@ export class SearchBookModal extends Component {
 
   handleSubmit = async () => {
 
+    console.log("actual submit")
```
Comment (steven-solar) [inline capstone-frontend/src/components/SearchBookModal.jsx]: `nit`: same as above

File: capstone-frontend/src/components/SearchBookModal.jsx [hunk 17]
```diff
@@ -107,20 +113,23 @@ export class SearchBookModal extends Component {
   }
 
   clubSubmit = (gbookID) => {
+
+    console.log("club submit")
```
Comment (steven-solar) [inline capstone-frontend/src/components/SearchBookModal.jsx]: `nit`: same as above

File: capstone-frontend/src/components/LeftSideBar.jsx (hunk 12 above)
Comment (steven-solar) [reply capstone-frontend/src/components/LeftSideBar.jsx:359]: I found myself struggling to open the right pages without seeing text

**Review by steven-solar — APPROVED** [review]
Nice job! This will give us much better responsive design!

File: capstone-frontend/src/components/ClubPage.jsx (hunk 10 above)
Comment (antoniolinhart) [reply capstone-frontend/src/components/ClubPage.jsx:69]: +1

File: capstone-frontend/src/components/CreateClub.jsx [hunk 18]
```diff
@@ -17,10 +17,11 @@ class CreateClub extends Component {
     }
     data.ownerID = window.localStorage.getItem("userID");
 
+    console.log("Create club", data)
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/CreateClub.jsx]: nit: remove console.log

File: capstone-frontend/src/components/CreateClub.jsx (hunk 11 above)
Comment (antoniolinhart) [reply capstone-frontend/src/components/CreateClub.jsx]: Did you mean `/clubpage/${club.id}`?

File: capstone-frontend/src/components/CreateClub.jsx (hunk 18 above)
Comment (MauroGuerrero20) [inline capstone-frontend/src/components/CreateClub.jsx]: nit: remove

File: capstone-frontend/src/styles/LeftSideBar.css [hunk 19]
```diff
@@ -1,3 +1,31 @@
...
+  color: inherit;
+  text-decoration: none;
+}
+
+.avatar-text-style {
+  font-size: 0.9rem !important;
+  font-weight: 400 !important;
+  margin-top: 0.4rem !important;
+  padding-left: 1rem;
+  padding-right: 1rem;
```
Comment (antoniolinhart) [inline capstone-frontend/src/styles/LeftSideBar.css]: nit: alphabetize styles and inner styles (color before cursor)

**Review by antoniolinhart — COMMENTED** [review]
I really like this! Especially the mobile changes make it really nice and clean!

File: capstone-frontend/src/components/CreateClub.jsx (hunk 11 above)
Comment (steven-solar) [reply capstone-frontend/src/components/CreateClub.jsx]: I did! Thanks for catching that. The URL should read the exact same as before, but replace data with club.

**Review by antoniolinhart — APPROVED** [review]
Meant to approve before.

File: capstone-frontend/src/App.jsx (hunk 6 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/App.jsx:40]: I tried to added in this PR. But it turn out it will be very hard and complex to add it. I don't think its worth spending the time trying to implement if its just a minor feature

File: capstone-frontend/src/components/BookDescriptionOverlay.jsx (hunk 9 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/BookDescriptionOverlay.jsx:9]: This check resolves issues @antoniolinhart and I were having with the Popover Overlay inside the SearchBookModal.

Comment (MauroGuerrero20) [reply capstone-frontend/src/components/BookDescriptionOverlay.jsx:9]: The search modal was send an undefined prop causing an error.

File: capstone-frontend/src/components/ClubPage.jsx (hunk 10 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx:69]: Yeah sorry, I didn't mean to format this file.

File: capstone-frontend/src/components/Logout.jsx (hunk 13 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Logout.jsx:23]: Well I may not want to apply a style. I think my LeftSideBar doesn't actually apply a style when it uses ```<Logout/>```. I just don't want to limit our self on styling.

File: capstone-frontend/src/components/ClubPage.jsx [hunk 20]
```diff
@@ -120,13 +122,13 @@ class ClubPage extends Component {
...
     return (
-      <div className='container text-center'> 
+      <div className='container text-center'>
         {isOwner &&
-           <Link to={`/adminclubpage/${this.state.club.id}`}> 
-              <Button className='admin-button' variant='secondary'>
-                Admin page
+          <Link to={`/adminclubpage/${this.state.club.id}`}>
+            <Button className='admin-button' variant='secondary'>
+              Admin page
```
Comment (VivekShankar) [inline capstone-frontend/src/components/ClubPage.jsx:129]: nit: Can you fix the alignment of the `Button` opening and closing tags?

File: capstone-frontend/src/components/SearchBookModal.jsx (hunk 15 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/SearchBookModal.jsx]: Done.

File: capstone-frontend/src/components/SearchBookModal.jsx (hunk 16 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/SearchBookModal.jsx]: Done.

File: capstone-frontend/src/components/SearchBookModal.jsx (hunk 17 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/SearchBookModal.jsx]: Done.

File: capstone-frontend/src/App.css (hunk 2 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/App.css]: Done.

File: capstone-frontend/src/App.test.jsx (hunk 3 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/App.test.jsx:33]: Correct. Added a comment explaining why.

File: capstone-frontend/src/components/CreateClub.jsx (hunk 18 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/CreateClub.jsx]: Done.

File: capstone-frontend/src/styles/LeftSideBar.css (hunk 19 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/styles/LeftSideBar.css]: Done.

File: capstone-frontend/src/components/CreateClub.jsx (hunk 11 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/CreateClub.jsx]: Done.

File: capstone-frontend/src/components/LeftSideBar.jsx (hunk 12 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/LeftSideBar.jsx:359]: We could change the icon later down the road. The point of this LeftSideBar is that is a MiniDrawer, so you can expanded and minimize it.

Comment (MauroGuerrero20) [reply capstone-frontend/src/components/LeftSideBar.jsx:359]: I don't think names would fit the style of the drawer. We could start the Drawer open by default if you want?

File: capstone-frontend/src/components/MyClubs.jsx [hunk 21]
```diff
@@ -67,14 +73,14 @@ class MyClubs extends Component {
         <Row>
           <Col xs={12} className="title"> My Clubs </Col>
         </Row>
-          <Link to="/createclub"> 
-            <Button variant="primary">
-              Create New Club 
+        <Link to="/createclub">
+          <Button variant="primary">
```
Comment (VivekShankar) [inline capstone-frontend/src/components/MyClubs.jsx:77]: nit: Can you fix the alignment of the Button opening and closing tags?

**Review by VivekShankar — APPROVED** [review]
LGTM

File: capstone-frontend/src/components/ClubPage.jsx (hunk 20 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx:129]: Done.

File: capstone-frontend/src/components/MyClubs.jsx (hunk 21 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/MyClubs.jsx:77]: Done.
